packages/accounts-base/accounts_server.js

```diff
@@ -68,18 +68,16 @@ export class AccountsServer extends AccountsCommon {
 
   // @override of "abstract" non-implementation in accounts_common.js
   userId() {
-    // This function only works if called inside a method. In theory, it
-    // could also be called from publish statements, since they also
-    // have a userId associated with them. However, given that publish
-    // functions aren't reactive, using any of the infomation from
-    // Meteor.user() in a publish function will always use the value
-    // from when the function first runs. This is likely not what the
-    // user expects. The way to make this work in a publish is to do
-    // Meteor.find(this.userId).observe and recompute when the user
-    // record changes.
-    var currentInvocation = DDP._CurrentInvocation.get();
+    // This function only works if called inside a method or a pubication.
+    // Using any of the infomation from Meteor.user() in a method or
+    // publish function will always use the value from when the function first
+    // runs. This is likely not what the user expects. The way to make this work
+    // in a method or publish function is to do Meteor.find(this.userId).observe
+    // and recompute when the user record changes.
+    var currentInvocation = DDP._CurrentPublicationInvocation.get();
+    currentInvocation = currentInvocation || DDP._CurrentMethodInvocation.get();
```

Comment:
I believe this would make sense as it's more likely that `DDP._CurrentMethodInvocation` would be set. If that isn't set, we're in a publish function and we fall back to `DDP._CurrentPublicationInvocation`.

So given the first comment I would change this to `const currentInvocation = DDP._CurrentMethodInvocation.get() || DDP._CurrentPublicationInvocation.get();`.